ui/src/lib/yjs/socketYjsManager.ts

```diff
@@ -0,0 +1,309 @@
+          content: { room_id: this.doc.clientID.toString() },
+        },
+      });
+
+      await this.sendFlowMessage({
+        event: {
+          tag: "Join",
+          content: { room_id: this.doc.clientID.toString() },
+        },
+      });
+
+      await this.sendFlowMessage({
+        event: {
+          tag: "Emit",
+          content: { data: "" },
+        },
+        session_command: {
+          tag: "Start",
+          content: {
+            project_id: this.projectId || "",
+            user: {
+              id: this.doc.clientID.toString(),
+              tenant_id: this.projectId,
+              name: "defaultName",
+              email: "defaultEmail@example.com",
+            },
+          },
+        },
+      });
+
+      await this.syncData();
+    } catch (error) {
+      console.error("Failed to initialize room:", error);
+    }
+  }
+
+  async isReady(): Promise<boolean> {
+    if (this.socketReady) return true;
+    await sleep(100);
+    return await this.isReady();
+  }
+
+  protected onReady() {
+    this.socketReady = true;
+  }
+
+  protected onPeerUpdate(data: { update: ArrayBuffer | Uint8Array }) {
+    const update = data.update instanceof ArrayBuffer
+      ? new Uint8Array(data.update)
+      : data.update;
+  
+    const currentState = Y.encodeStateAsUpdateV2(this.doc);
+    const diffUpdate = Y.diffUpdateV2(update, currentState);
+    Y.applyUpdateV2(this.doc, diffUpdate, 'peer');
+    this.onUpdateHandlers.forEach((handler) => handler(update));
+  }
+
+  async syncData() {
+    await this.isReady();
+
+    const currentState = Y.encodeStateAsUpdateV2(this.doc);
+    const stateVector = Y.encodeStateVectorFromUpdateV2(currentState);
+    if (this.ws.readyState === WebSocket.OPEN) {
+      this.ws.send(stateVector);
+    }
+
+    if (!this.firstSyncComplete) {
+      this.firstSyncComplete = true;
+      queueMicrotask(() => this.syncData());
+    }
+  }
+
+  private async sendFlowMessage(message: FlowMessage): Promise<void> {
+    return new Promise((resolve, reject) => {
+      if (this.ws.readyState !== WebSocket.OPEN) {
+        reject(new Error("WebSocket is not connected"));
+        return;
+      }
+
+      try {
+        this.ws.send(JSON.stringify(message));
+        resolve();
+      } catch (error) {
+        reject(error);
+      }
+    });
+  }
+
+  protected onDocUpdate(update: Uint8Array, origin: unknown) {
+    if (origin === this.doc.clientID && this.ws.readyState === WebSocket.OPEN) {
+      const stateVector = Y.encodeStateVectorFromUpdateV2(update);
+      const diffUpdate = Y.diffUpdateV2(update, stateVector);
+      
+      this.sendFlowMessage({
+        event: {
+          tag: "Emit",
+          content: { data: "" }
+        },
+        session_command: {
+          tag: "MergeUpdates",
+          content: {
+            project_id: this.projectId || "",
+            data: new Uint8Array(diffUpdate),
+            updated_by: this.doc.clientID.toString()
```

Comment:
Is this used internally only for the websocket server to be able to resolve conflicts between session users? If we want to be able to reference this across sessions, we don't want the clientID, which I believe is auto-generated on each new session connection. We would want the user id probably.